encoding.bs

```diff
@@ -1236,42 +1246,38 @@ string += decoder.decode(); // end-of-stream</code></pre>
 </dl>
 
 <p>The
-<dfn constructor for=TextDecoder id=dom-textdecoder><code>TextDecoder(<var>label</var>, <var>options</var>)</code></dfn>
-constructor, when invoked, must run these steps:
+<dfn constructor for=TextDecoder lt="TextDecoder(label, options)" id=dom-textdecoder><code>new TextDecoder(<var>label</var>, <var>options</var>)</code></dfn>
+constructor steps are:
 
 <ol>
  <li><p>Let <var>encoding</var> be the result of <a>getting an encoding</a> from <var>label</var>.
 
  <li><p>If <var>encoding</var> is failure or <a>replacement</a>, then <a>throw</a> a {{RangeError}}.
 
- <li><p>Let <var>dec</var> be a new {{TextDecoder}} object.
-
- <li><p>Set <var>dec</var>'s <a for=TextDecoderCommon>encoding</a> to <var>encoding</var>.
+ <li><p>Set <a>this</a>'s <a for=TextDecoderCommon>encoding</a> to <var>encoding</var>.
 
- <li><p>If <var>options</var>'s <code>fatal</code> member is true, then set <var>dec</var>'s
+ <li><p>If <var>options</var>["<code>fatal</code>"] is true, then set <a>this</a>'s
  <a for=TextDecoderCommon>error mode</a> to "<code>fatal</code>".
 
- <li>If <var>options</var>'s <code>ignoreBOM</code> member is true, then set <var>dec</var>'s
- <a for=TextDecoderCommon>ignore BOM</a> to true.
-
- <li><p>Return <var>dec</var>.
+ <li><p>Set <a>this</a>'s <a for=TextDecoderCommon>ignore BOM</a> to
+ <var>options</var>["<code>ignoreBOM</code>"].
 </ol>
 
 <p>The <dfn method for=TextDecoder><code>decode(<var>input</var>, <var>options</var>)</code></dfn>
-method, when invoked, must run these steps:
+method steps are:
 
 <ol>
- <li><p>If <a for=TextDecoder>do not flush</a> is false, set <a for=TextDecoder>decoder</a>
- to a new <a for=TextDecoderCommon>encoding</a>'s <a for=/>decoder</a>,
- <a for=TextDecoder>stream</a> to a new <a for=/>stream</a>, and
- <a for=TextDecoderCommon>BOM seen</a> to false.
+ <li><p>If <a>this</a>'s <a for=TextDecoder>do not flush</a> is false, then set <a>this</a>'s
+ <a for=TextDecoder>decoder</a> to a new <a for=TextDecoderCommon>encoding</a>'s
```

Comment:
This for is wrong (preexisting problem)

Reply:
What is wrong about it?

Reply:
The for=TextDecoderCommon should be for=/

Reply:
Why?

Reply:
I'm not sure how that would make sense.

Reply:
You're right, that wouldn't work. Instead this should match the `new TextDecoderStream` step 6: "Set this’s decoder to a new decoder for this’s encoding"